### PR TITLE
Rename OAuthAuthentication* to OAuth* and Authenticated to AccessTokenReceived.

### PR DESCRIPTION
**Pull request by Tratcher**
#98
@HaoK @PinpointTownes @Eilon @brockallen

Note I did not rename the Handler, Middleware, or Options classes because A) the type names are not referenced directly in user code, and B) I didn't want to confuse the inheritance heirarchy for derived classes: FacebookAuthenticationOptions -> OAuthAuthenticationOptions -> AuthenticationOptions

I also cleaned out some obsolete Context classes.

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Semantically, this naming change is totally appropriate: OAuth2 is an authorization protocol (though we use it for pseudo-authentication).

Technically, I'm not sure it's really pertinent: `OAuthAuthenticationMiddleware` is an authentication middleware, no matter what we think. It's derived from `AuthenticationMiddleware`, it calls `SignInAsync` during the callback process and the user is considered as authenticated after a complete "authorization flow", even if you don't provide an explicit `AccessTokenReceived` notification. 

IMHO, removing `Authentication` from the extension name will just make things less clear.

File: test/Microsoft.AspNet.Authentication.Test/Facebook/FacebookMiddlewareTests.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public class FacebookMiddlewareTests
                     {
                         options.AppId = "Test App Id";
                         options.AppSecret = "Test App Secret";
-                        options.Notifications = new OAuthAuthenticationNotifications
+                        options.Notifications = new OAuth.OAuthNotifications
```
Comment (PinpointTownes) [inline test/Microsoft.AspNet.Authentication.Test/Facebook/FacebookMiddlewareTests.cs]: Hum, why do you need `OAuth.` here?

**Comment by HaoK** [comment]
Or Maybe we should shorten everything to auth?

UseGoogleAuth
UseFacebookAuth

**Comment by PinpointTownes** [comment]
Or **app.UseOAuthAuthenrization** :trollface:

**Comment by HaoK** [comment]
:shipit:

**Comment by Tratcher** [comment]
So just close this and leave everything as is?

**Comment by Tratcher** [comment]
Aborted

**Pull request closed without merging**